src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -413,6 +401,42 @@ internal static class RuntimeHelpers
                 .CreateDelegate<Action>();
         }
 
+        internal static class StackTrace
+        {
+            internal const string TypeName = "System.Diagnostics.StackTrace";
+
+            internal static readonly Type Type = ReflectionUtilities.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Diagnostics_StackTrace}",
+                desktopName: TypeName);
+
+            private static readonly ConstructorInfo s_Ctor = Type != null
+                ? Type.GetTypeInfo().GetDeclaredConstructor(new Type[] { })
+                : null;
+
+
+            private static readonly MethodInfo s_ToString = Type != null
+                ? Type.GetTypeInfo().GetDeclaredMethod("ToString", new Type[] { })
+                : null;
+
+            internal static string GetString()
+            {
+                const string StackTraceUnavailable = "StackTrace unavailable.";
+
+                if (s_ToString == null)
```

Comment:
This should never be `null`.  The `ToString` contract is supported on all platforms including ProjectK and ProjectN.  We should be safe here to use this as we do other guaranteed to be present methods.